src/app/state/persist/saga.ts

```diff
@@ -211,8 +219,55 @@ function* encryptAndPersistState(action: AnyAction) {
   window.localStorage.setItem(STORAGE_FIELD, encryptedState)
 }
 
+function* retainEncryptionKeyBetweenPopupReopenings() {
+  if (runtimeIs !== 'extension') return
+  yield* fork(function* () {
+    const channelQueue = yield* actionChannel<AnyAction>('*')
+    let previousStringifiedEncryptionKey: PersistState['stringifiedEncryptionKey'] = undefined
+    while (true) {
+      yield* take(channelQueue)
+      const stringifiedEncryptionKey = yield* select(selectStringifiedEncryptionKey)
+      if (stringifiedEncryptionKey !== previousStringifiedEncryptionKey) {
+        previousStringifiedEncryptionKey = stringifiedEncryptionKey
+        yield* call(writeSharedExtMemory, stringifiedEncryptionKey)
+      }
+    }
+  })
+
+  yield* fork(function* () {
+    const encryptedState = window.localStorage.getItem(
+      STORAGE_FIELD,
+    ) as EncryptedString<PersistedRootState> | null
+    if (!encryptedState) return // Ignore
+    try {
+      const stringifiedEncryptionKey = yield* call(readSharedExtMemory)
+      if (!stringifiedEncryptionKey) return // Ignore
+      if (stringifiedEncryptionKey === 'skipped') return // Ignore
+      const keyWithSalt: KeyWithSalt = fromBase64andParse(stringifiedEncryptionKey)
+      const persistedRootState = yield* call(decryptWithKey<PersistedRootState>, keyWithSalt, encryptedState)
+      yield* put(persistActions.setUnlockedRootState({ persistedRootState, stringifiedEncryptionKey }))
+    } catch (error) {
+      // Ignore
+    }
+  })
+}
+
+async function writeSharedExtMemory(stringifiedEncryptionKey: PersistState['stringifiedEncryptionKey']) {
+  if (runtimeIs !== 'extension') return
+  const browser = await import('webextension-polyfill')
+  await browser.storage.session.set({ stringifiedEncryptionKey })
+}
+
+async function readSharedExtMemory() {
+  if (runtimeIs !== 'extension') return
+  const browser = await import('webextension-polyfill')
+  const storage = await browser.storage.session.get('stringifiedEncryptionKey')
+  return storage.stringifiedEncryptionKey as PersistState['stringifiedEncryptionKey']
+}
+
 export function* persistSaga() {
   yield* watchPersistAsync()
+  yield* retainEncryptionKeyBetweenPopupReopenings()
```

Comment:
Do we need to take into account `unlockedStatus` in this logic? 

For example:
- I am locking my wallet using Profile dropdown -> Profile -> Lock Profile
- Close extension
- Open popup - wallet is open (no UnlockGate)

Reply:
Nope, should already have been handled, but chrome has a bug https://github.com/w3c/webextensions/issues/263
so `await browser.storage.session.set({ stringifiedEncryptionKey })` does nothing when undefined

Reply:
Workaround in https://github.com/oasisprotocol/wallet/pull/2120/commits/db49e882699d6e851b46b31274031bd31d1b3f38